Fix warning in compilation

After the PR #4 was merged now we have the following warning

```
[1/2] Compiling C object kc_validator.so.p/src_kc_validator.c.o
../src/kc_validator.c: In function ‘kc_decision’:
../src/kc_validator.c:497:8: warning: ‘post’ may be used uninitialized [-Wmaybe-uninitialized]
  497 |     if (post) pfree(post); /* post might not be set, but pfree(NULL) is safe */
      |        ^
../src/kc_validator.c:344:11: note: ‘post’ was declared here
  344 |     char *post;
      |           ^~~~
[2/2] Linking target kc_validator.so
```

--- a/src/kc_validator.c
+++ b/src/kc_validator.c
@@ -338,10 +338,10 @@ kc_decision(CURL *curl, const char *permission, const char *user_token)
     char errbuf[CURL_ERROR_SIZE] = {0};
     long http_code;
     double total_time;
-    char *aud;
-    char *perm_enc;
-    char *cid;
-    char *post;
+    char *aud = NULL;
+    char *perm_enc = NULL;
+    char *cid = NULL;
+    char *post = NULL;
     long connect_tmo;
 
     /* Initialize response buffer (must be pfree'd) */

